### PR TITLE
Upstream 0.16.1

### PR DESCRIPTION
**Pull request by Fauzyy**
This PR updates our running fork to the latest upstream version.

This PR is huge, the best way to review would be to look at the commits at the bottom, from `
b18406e` onward

I tested this locally with ngrok hooked up to terracode-sandbox. Next steps would be to cut a release and pull that into atlantis-system and do more testing onstaging.

### CODE REVIEW

File: server/events/project_command_pool_executor.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ func runProjectCmdsParallel(
 	var results []models.ProjectResult
 	mux := &sync.Mutex{}
 
-	wg := sizedwaitgroup.New(15)
+	wg := sizedwaitgroup.New(poolSize)
```
Comment (eculver) [inline server/events/project_command_pool_executor.go:20]: do we need to pull in this configuration somewhere to make sure we're creating a big enough pool?

**Review by eculver — APPROVED** [review]
This is, in fact, a monster, but the few changes of yours on top of the rebase are pretty friendly. Just one comment. Vayan con dios amigo 🖖

File: server/events/project_command_pool_executor.go (hunk 1 above)
Comment (Fauzyy) [reply server/events/project_command_pool_executor.go:20]: We have this set to 15 right now through a config flag that's pass in as a env var from chamber. For some reason they hard coded this to 15 upstream, even though they support a config flag to change it 😖  - I think it was an error

Comment (Fauzyy) [reply server/events/project_command_pool_executor.go:20]: I can open a PR upstream to fix this later

**Pull request merged**